test/BF16/matmul-pbf16.mlir

```diff
@@ -15,26 +15,28 @@ func.func @matmultpp(%A: memref<4x8xbf16>,
 func.func @entry() {
   %c0 = arith.constant 0 : index
   %f0 = arith.constant 1.0 : bf16
-  %da = memref.alloc() :memref<4x8xbf16>
+  
+  %da = memref.alloc() : memref<4x8xbf16>
   linalg.fill ins(%f0 : bf16) outs(%da : memref<4x8xbf16>)
-  %db = memref.alloc() :memref<8x4xbf16>
-  linalg.fill ins(%f0:bf16) outs (%db:memref<8x4xbf16>)
+  
+  %db = memref.alloc() : memref<4x4x2xbf16>
```

Comment:
This test was testing the packing, now it isn't any more. We really shouldn't remove that functionality from these tests.

Reply:
Looking at the passes in the test I am not sure the intent was to test the packing pass, there is no pack. It is written as IR and check e2e. Integration for pack is added in another test.

Reply:
It had an explicit `pack` before a `matmul`, which was testing the VNNI packing.

Sure we can replace that with an already "packed" tensor, but then this wouldn't be the same test.

Reply:
This test just checks VNNI matmul results for packed layout, it just so happens that I am using pack operation to transform the layout. So, its fine to use the buffer in packed shape directly.